test/test_environments/test_state.cpp

```diff
@@ -0,0 +1,36 @@
+#include <State.h>
+#include <gtest/gtest.h>
+#include <string>
+#include <vector>
+
+struct BasicStruct {
+    BasicStruct(int val, std::string str1) : intVal(val), stringVal(str1) {}
+
+  public:
+    std::string stringVal;
+    int intVal;
+};
+
+TEST(State, TestVariableInput) {
+    // Arange
+    std::vector<int> someVec = {1, 2, 3, 4};
+    BasicStruct someStruct = BasicStruct(10, "test");
+
+    // Act and Assert
+    EXPECT_NO_FATAL_FAILURE(State(1));
+    EXPECT_NO_FATAL_FAILURE(State("1"));
+    EXPECT_NO_FATAL_FAILURE(State vecState = State(someVec));
+    EXPECT_NO_FATAL_FAILURE(State structState = State(someStruct));
+}
+
+TEST(State, TestInputHandling) {
```

Comment:
Rename test to something more descriptive like "TestStateInputOutputEqual" maybe?